core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java

```diff
@@ -195,7 +195,8 @@ private InnerHitBuilder(InnerHitBuilder other) {
         }
     }
 
-    InnerHitBuilder(InnerHitBuilder other, QueryBuilder query, String parentChildType, boolean ignoreUnmapped) {
+    // TODO public for hasChild and hasParent query.
+    public InnerHitBuilder(InnerHitBuilder other, QueryBuilder query, String parentChildType, boolean ignoreUnmapped) {
```

Comment:
maybe add a note that these methods are for internal usage only and no one should rely on this method to remain to be available?